Bump min supported WC version to 6.9 and update few DEWP & dev dependencies

tomalec opened the pull request:
### Changes proposed in this Pull Request:

- [Bump min supported WC version to 6.9](https://github.com/woocommerce/google-listings-and-ads/commit/3e371ae7178f351ed8eb171cc917903e01b072e7)

   To follow L-2
- [Update `@woocommerce/customer-effort-score` to the 2.2.0 used by WC `6.9.4`](https://github.com/woocommerce/google-listings-and-ads/commit/69f212497c19593be6ef06f7b8cd1184d94764ec) 
- [Update `@woocommerce/navigation` to the 8.1.0 used by WC 6.9.4](https://github.com/woocommerce/google-listings-and-ads/commit/666106e46ee56f6434a6a5e5017a6fd2a2a4e40b)
- [Update `@wordpress/scripts@24.6.0`](https://github.com/woocommerce/google-listings-and-ads/commit/68228872ad746a79c7876ef9d7f4d7187548b516)
- [Remove lint:md:js scripts as it's no longer in wp-scripts](https://github.com/woocommerce/google-listings-and-ads/commit/edceea0978b9097a1bc43398c486c7120842485c)
- [Make our `package.json` compliant with lint-pkg-json](https://github.com/woocommerce/google-listings-and-ads/commit/74449ac640e03ba60eed96f3b97511d3cadeb304)


### Screenshots:

<!--- Optional --->


### Detailed test instructions:
<!-- Add detailed instructions for how to test that this PR fixes the issue and confirm that it doesn't break any other features :) -->

1. Checkout branch
```
npm i
composer install
npm run build
npm run test:js
npm run lint
npm run lint-pkg-json
```

### Additional details:
1. `lint-pkg-json` still complains about the license, hopefully, eventually it may be fixed by https://github.com/WordPress/gutenberg/pull/45823
2. There are two packages that should be updated: `@woocommerce/data@4.1.0` & `@woocommerce/components@10.3.0` as they lag far behind WC 6.6, but those updates would require tweaking with Jest config as in https://github.com/woocommerce/google-listings-and-ads/pull/1752

<!--
Optional.
Enter a summary of all changes in this Pull Request, which will be added to the changelog if accepted.
Each line should start with change type prefix`(Fix|Add|…) - `, for example:
> Break - A change breaking previous API or functionality.
> Add - A new feature, function or functionality was added.
> Update - Big changes to something that wasn't broken.
> Fix - Took care of something that wasn't working.
> Tweak - Small change, that isn't actually very important.
> Dev - Developer-facing only change.
> Doc - Updated customer or developer facing documentation

If you remove the "Changelog entry" header, the Pull Request title will be used as the changelog entry.

Add the `changelog: none` label if no changelog entry is needed.
-->
### Changelog entry

> Update - Drop support for WC < 6.9.

tomalec committed:
Bump min supported WC version to 6.9

tomalec committed:
Update `@woocommerce/customer-effort-score` to the `2.2.0` used by WC `6.9.4`

tomalec committed:
Update `@woocommerce/navigation` to the `8.1.0` used by WC `6.9.4`

tomalec committed:
Update `@wordpress/scripts@24.6.0`

tomalec committed:
Remove `lint:md:js` scripts as it's no longer in wp-scripts

tomalec committed:
Make our `package.json` compliant with `lint-pkg-json`

eason9487 committed:
Merge branch 'develop' into dev/bump-dewped

eason9487 commented:
I merged `develop` to this PR since I'm trying to resolve the JS unit test issue that occurred in #1752. Based on this PR would be smoother because it needs to bump min supported WC as well and include recreating package-lock.json.

eason9487 reviewed (approved):
Thanks for updating the min WC version and upgrading the dependencies. Testing well. LGTM. An npm script could be adjusted to make the semantics of its script name more consistent but this is not a blocker.

---

Just realized `wp-scripts format` is not only formatting JavaScript-related files but also YAML files. It included JSON files since [version 23.1.0](https://github.com/WordPress/gutenberg/blob/trunk/packages/scripts/CHANGELOG.md#2310-2022-05-18). Maybe we could change the `format:js` script to:
1. `"format:js": "wp-scripts format \"./**/*.js\""`
2. or `"format": "wp-scripts format"`

tomalec committed:
Rename `format:js` to `format` to match the behavior.

Addresses https://github.com/woocommerce/google-listings-and-ads/pull/1775#pullrequestreview-1185957123

Related to
https://github.com/WordPress/gutenberg/blob/trunk/packages/scripts/CHANGELOG.md#2310-2022-05-18

tomalec committed:
Merge branch 'dev/bump-dewped' of github.com:woocommerce/google-listings-and-ads into dev/bump-dewped
